algorithms/src/std_algorithms/impl/Kokkos_PartitionCopy.hpp

```diff
@@ -126,6 +125,55 @@ partition_copy_impl(const std::string& label, const ExecutionSpace& ex,
           to_first_false + counts.false_count_};
 }
 
+template <class TeamHandleType, class InputIteratorType,
+          class OutputIteratorTrueType, class OutputIteratorFalseType,
+          class PredicateType>
+KOKKOS_FUNCTION ::Kokkos::pair<OutputIteratorTrueType, OutputIteratorFalseType>
+partition_copy_team_impl(const TeamHandleType& teamHandle,
+                         InputIteratorType from_first,
+                         InputIteratorType from_last,
+                         OutputIteratorTrueType to_first_true,
+                         OutputIteratorFalseType to_first_false,
+                         PredicateType pred) {
+  // impl uses a scan, this is similar how we implemented copy_if
+
+  // checks
+  Impl::static_assert_random_access_and_accessible(
+      teamHandle, from_first, to_first_true, to_first_false);
+  Impl::static_assert_iterators_have_matching_difference_type(
+      from_first, to_first_true, to_first_false);
+  Impl::expect_valid_range(from_first, from_last);
+
+  if (from_first == from_last) {
+    return {to_first_true, to_first_false};
+  }
+
+  const std::size_t num_elements =
+      Kokkos::Experimental::distance(from_first, from_last);
+
+  // FIXME: there is no parallel_scan overload that accepts TeamThreadRange and
+  // return_value, so temporarily serial implementation is used
```

Comment:
This is changing with recent pull requests, though.

Reply:
yes, i am going to fix all of this *after* all the team PRs are merged in a follow up.